### PR TITLE
APPSERV-18 Test for testing get application name

### PR DESCRIPTION
**Pull request by MeroRai**
# Description
This PR adds unit tests for this PR: https://github.com/payara/Payara/pull/4384

### Testing Environment
Zulu JDK 1.8_222 on Elementary OS 0.4.1 Loki with Maven 3.5.4>

### CODE REVIEW

**Comment by MarkWareham** [comment]
Without deeply diving into the code, I would suggest some tests when USER_PROVIDED_APP_NAME has not been provided (if that's possible)

**Comment by MeroRai** [comment]
@jbee I am testing the behaviour of `genericHandler.getDefaultApplicationName` method. I have mocked the behaviour of its dependencies so it doesn't rely on them.

**Comment by jbee** [comment]
@MeroRai  Missed that the stub extends the implementation class.

File: nucleus/common/internal-api/src/test/java/org/glassfish/internal/deployment/GetApplicationNameTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @Test
+    public void testWhenDDDoesNotHaveAppName() {
+        when(applicationInfoProvider.getNameFor(readableArchive, null)).thenReturn(null);
+        assertEquals(USER_PROVIDED_APP_NAME, genericHandler.getDefaultApplicationName(readableArchive, null, USER_PROVIDED_APP_NAME));
+    }
+
+    @Test
+    public void testWhenDDAppNameIsSetToDefaultArchiveName() {
+        when(applicationInfoProvider.getNameFor(readableArchive, null)).thenReturn(DEFAULT_ARCHIVE_NAME);
+        when(readableArchive.getName()).thenReturn(DEFAULT_ARCHIVE_NAME);
```
Comment (jbee) [inline nucleus/common/internal-api/src/test/java/org/glassfish/internal/deployment/GetApplicationNameTest.java]: I feel like a `assertSomething` helper method should be extended and used by each of the 3 test methods which should make them more readable in the the way: this input has that expected output. It should also become clear why only this text needs mockup of the `readableArchive.getName()`.

**Comment by MeroRai** [comment]
jenkins test please

**Pull request merged**